pinot-core/src/main/java/com/linkedin/pinot/core/indexsegment/IndexSegment.java

```diff
@@ -20,57 +20,53 @@
 import com.linkedin.pinot.core.startree.StarTree;
 
 
-/**
- * This is the interface of index segment. The index type of index segment
- * should be one of the supported {@link com.linkedin.pinot.core.indexsegment.IndexType
- * IndexType}.
- *
- *
- */
 public interface IndexSegment {
-  /**
-   * @return
-   */
-  public IndexType getIndexType();
 
   /**
-   * @return
+   * Returns the name of the segment.
+   *
+   * @return Segment name
    */
-  public String getSegmentName();
+  String getSegmentName();
 
   /**
-   * @return
+   * Returns the {@link SegmentMetadata} of the segment.
+   *
+   * @return Segment metadata
    */
-  public String getAssociatedDirectory();
+  SegmentMetadata getSegmentMetadata();
 
   /**
-   * @return SegmentMetadata
+   * Returns all the columns inside the segment.
+   *
+   * @return Array of column names
    */
-  public SegmentMetadata getSegmentMetadata();
+  String[] getColumnNames();
```

Comment:
Does this require implementations to convert their storage (eg List/Set) into a new String[], creating unnecessary garbage? If so, let's see if we can cleanup the interface to appropriate return type.